proto/contract.proto

```diff
@@ -11,6 +11,48 @@ option java_outer_classname = "DataPlaneContract";
 // Protobuf include nightmare? No thanks!
 message Empty {}
 
+message Exact {
+  map<string, string> attributes = 1;
+}
+
+message Prefix {
+  string attribute = 1;
+  string prefix = 2;
+}
+
+message Suffix {
+  string attribute = 1;
+  string suffix = 2;
+}
```

Comment:
I'm reading [this example in spec](https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#filters):
```
[ {"prefix": { "type": "com.github.issue" } } ]
```

why doesn't this need to be a `map<string, string>`?

In addition, at this layer, I would design for extensibility here.

Reply:
I see why:
> Use of this MUST include exactly one nested property

It seems weird that exact supports multiple attributes while prefix and suffix don't